graphjet-core/src/main/java/com/twitter/graphjet/algorithms/TweetAuthorFilter.java

```diff
@@ -65,6 +68,6 @@ public void resetFilter(RecommendationRequest request) {}
    */
   @Override
   public boolean filterResult(long resultNode, SmallArrayBasedLongToDoubleMap[] socialProofs) {
-    return !authoredTweets.contains(resultNode);
+    return hasAuthors && !authoredTweets.contains(resultNode);
```

Comment:
That's exactly what the boolean variable does, i.e. to preserve whether authors are empty. tweetAuthors only exists in the scope of the object constructor, so we must have some sort of additional variable to store this information.